File: 0090-subsets-ii/0090-subsets-ii.java
class Solution {
    public List<List<Integer>> subsetsWithDup(int[] nums) {
        Arrays.sort(nums);
        List<List<Integer>> ans = new ArrayList<>();
        SWD(0, new ArrayList<>(), ans, nums);
        return ans;
    }
    private void SWD(int indx, List<Integer> ds, List<List<Integer>> ans, int[] n) {
        ans.add(new ArrayList<>(ds));
        for(int i = indx; i < n.length; i++) {
            if (i > indx && n[i] == n[i-1]) continue;
            ds.add(n[i]);
            SWD(i+1, ds, ans, n);
            ds.remove(ds.size() - 1);
        }
    }
}

File: 0090-subsets-ii/0090-subsets-ii.cpp
class Solution {
public:
    vector<vector<int>> ans;
    void helper(int start, vector<int> nums, vector<int> path) {
        ans.push_back(path);
        for (int i = start; i < nums.size(); i++) {
            if (i > start && nums[i] == nums[i-1]) continue;
            path.push_back(nums[i]);
            helper(i+1, nums, path);
            path.pop_back();
        }
    }
    vector<vector<int>> subsetsWithDup(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        vector<int> path;
        helper(0, nums, path);
        return ans;
    }
};